internal/dataplane/parser/translators/httproute.go

```diff
@@ -350,14 +352,48 @@ func mustMarshalJSON[T any](val T) string {
 	return string(key)
 }
 
-// GeneratePluginsFromHTTPRouteFilters converts HTTPRouteFilter into Kong plugins.
+// SetRoutePlugins converts HTTPRouteFilter into Kong plugins. The plugins are set into the given kongstate.Route.
+// The plugins can be set in two different ways:
+// - Direct conversion from the respective HTTPRouteFilter.
+// - ExtensionRef to plugins annotation from the ExtensionRef filter.
+func SetRoutePlugins(route *kongstate.Route, filters []gatewayapi.HTTPRouteFilter, path string, tags []*string) error {
+	plugins, pluginAnnotation, err := generatePluginsFromHTTPRouteFilters(filters, path, tags)
+	if err != nil {
+		return err
+	}
+	if len(plugins) > 0 {
+		if route.Plugins == nil {
+			route.Plugins = make([]kong.Plugin, 0)
+		}
+		route.Plugins = append(route.Plugins, plugins...)
+	}
+	if len(pluginAnnotation) > 0 {
+		if route.Ingress.Annotations == nil {
+			route.Ingress.Annotations = make(map[string]string)
+		}
+		if _, ok := route.Ingress.Annotations[annotations.AnnotationPrefix+annotations.PluginsKey]; !ok {
+			route.Ingress.Annotations[annotations.AnnotationPrefix+annotations.PluginsKey] = pluginAnnotation
+		} else {
+			route.Ingress.Annotations[annotations.AnnotationPrefix+annotations.PluginsKey] = fmt.Sprintf("%s,%s",
+				route.Ingress.Annotations[annotations.AnnotationPrefix+annotations.PluginsKey],
+				pluginAnnotation)
+		}
+	}
+	return nil
+}
+
+// generatePluginsFromHTTPRouteFilters converts HTTPRouteFilter into Kong plugins.
 // path is the parameter to be used by the redirect plugin, to perform redirection.
-func GeneratePluginsFromHTTPRouteFilters(filters []gatewayapi.HTTPRouteFilter, path string, tags []*string) []kong.Plugin {
+// It returns two values:
+// - A set of plugin, generated by the conversion of all the filters but ExtensionRef.
```

Comment:
I'm not sure I understand this. Does that mean that extension refs are not converted to plugins 🤔 ? Shouldn't we in this case say something like:

```suggestion
// - A set of plugins generated by the conversion of all the provided filters, excluding  ExtensionRefs. Those are taken as is.
```

Feel free to change it.

Reply:
yeah, that's the intention. Changed